app/Silex/CultureFeed/CultureFeedServiceProvider.php

```diff
@@ -4,28 +4,33 @@
 
 namespace CultuurNet\UDB3\Silex\CultureFeed;
 
-use Silex\Application;
-use Silex\ServiceProviderInterface;
+use CultureFeed_DefaultOAuthClient;
+use CultuurNet\UDB3\Container\AbstractServiceProvider;
 
-final class CultureFeedServiceProvider implements ServiceProviderInterface
+final class CultureFeedServiceProvider extends AbstractServiceProvider
 {
-    public function register(Application $app)
+    protected function getProvidedServiceNames(): array
     {
-        $app['culturefeed'] = $app->share(
-            function (Application $app) {
-                $oauthClient = new \CultureFeed_DefaultOAuthClient(
-                    $app['culturefeed.consumer.key'],
-                    $app['culturefeed.consumer.secret']
+        return [
+            'culturefeed',
+        ];
+    }
+
+    public function register(): void
+    {
+        $container = $this->getContainer();
+
+        $container->addShared(
+            'culturefeed',
+            function () use ($container): CultureFeed_DefaultOAuthClient {
+                $oauthClient = new CultureFeed_DefaultOAuthClient(
+                    $container->get('config')['uitid']['consumer']['key'],
+                    $container->get('config')['uitid']['consumer']['secret'],
                 );
-                $oauthClient->setEndpoint($app['culturefeed.endpoint']);
+                $oauthClient->setEndpoint($container->get('config')['uitid']['base_url']);
 
-                return new \CultureFeed($oauthClient);
+                return $oauthClient;
```

Comment:
This still needs to be wrapped in `new \CultureFeed()`.

(Some context: https://github.com/cultuurnet/culturefeed-php/ is an old PHP library that existed before UiTdatabank v3, which we used as an SDK to communicate with UiTdatabank v2, and also UiTPAS and UiTID. These days it's only still used in a handful of places in this app)

Reply:
Damn, can't believe I missed that. Nice catch!

Reply:
It's understandable, it should become more obvious once we replace those string literals with FQNs of the classes that we register :)